v3/src/components/case-table/case-table-tool-shelf-button.tsx

```diff
@@ -1,29 +1,195 @@
+  const document = appState.document
+  const content = document.content
+  const manager = getSharedModelManager(document)
+  const datasets = manager?.getSharedModelsByType<typeof SharedDataSet>(kSharedDataSetType)
+  const existingTableTiles = document.content?.getTilesOfType(kCaseTableTileType)
+  const tileIds: string[] = []
+  existingTableTiles?.forEach(tile => tileIds.push(tile.id))
+  const row = content?.getRowByIndex(0)
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const [modalOpen, setModalOpen] = useState(false)
+  const [dataSetIdToDeleteId, setDataSetIdToDelete] = useState("")
+
+  if (!row) return null
+
+  const handleCreateNewDataSet = () => {
+    const tile = createDefaultTileOfType(kCaseTableTileType)
+    if (!tile) return
+    const newData = [{ AttributeName: "" }]
+    const ds = DataSet.create({ name: "New Dataset" })
+    ds.addAttribute({ name: "AttributeName" })
+    ds.addCases(toCanonical(ds, newData))
+    const { sharedData, caseMetadata } = gDataBroker.addDataSet(ds, tile.id)
+    manager?.addTileSharedModel(tile.content, sharedData, true)
+    manager?.addTileSharedModel(tile.content, caseMetadata, true)
+
+    const width = kDefaultTableSize.width
+    const height = kDefaultTableSize.height
+    const {x, y} = getPositionOfNewComponent({width, height})
+    content?.insertTileInRow(tile, row, {x, y, width, height})
+    uiState.setFocusedTile(tile.id)
+  }
+
+  const openTableForDataset = (model: ISharedDataSet, caseMetadata: ISharedCaseMetadata) => {
+    const tile = createDefaultTileOfType(kCaseTableTileType)
+      if (!tile) return
+      manager?.addTileSharedModel(tile.content, model, true)
+      manager?.addTileSharedModel(tile.content, caseMetadata, true)
+
+      const width = kDefaultTableSize.width
+      const height = kDefaultTableSize.height
+      const {x, y} = getPositionOfNewComponent({width, height})
+      content?.insertTileInRow(tile, row, {x, y, width, height})
+      uiState.setFocusedTile(tile.id)
+  }
 
+  const handleOpenDataSetTable = (dataset: ISharedDataSet) => {
+    const model = manager?.getSharedModelsByType("SharedDataSet").find(m =>  m.id === dataset.id) as ISharedDataSet
+    const caseMetadatas = manager?.getSharedModelsByType("SharedCaseMetadata") as ISharedCaseMetadata[]
+    const caseMetadata = caseMetadatas.find(cm => cm.data?.id === model.dataSet.id)
+    if (!model || !caseMetadata) return
+    if (existingTableTiles?.length !== 0) {
+      const tiles = manager?.getSharedModelTiles(model)
+      const tableTile = tiles?.find(tile => tile.content.type === "CodapCaseTable")
+      if (tableTile) { // a case table for the data set already exists in document
+        uiState.setFocusedTile(tableTile.id)
+      } else { // open a table and add it to the shared model manager
+        openTableForDataset(model, caseMetadata)
+      }
+    } else {
+      openTableForDataset(model, caseMetadata)
+    }
```

Comment:
Isn't this the same as this simpler version:
```suggestion
    if (existingTableTiles?.length !== 0) {
      const tiles = manager?.getSharedModelTiles(model)
      const tableTile = tiles?.find(tile => tile.content.type === "CodapCaseTable")
      if (tableTile) { // a case table for the data set already exists in document
        uiState.setFocusedTile(tableTile.id)
        return
      }
      // open a table and add it to the shared model manager
      openTableForDataset(model, caseMetadata)
    }
```

Reply:
The first `else` allows for more than one table to be open in the document for multiple data sets.
The second `else` prevents multiple tables for the same data set to be opened.

Reply:
I'm not trying to change the logic; I just think the code can be simplified while still achieving the same result. Your confusion is warranted, however, since my first suggestion wasn't correct. I think this should be equivalent:
```suggestion
    if (existingTableTiles?.length !== 0) {
      const tiles = manager?.getSharedModelTiles(model)
      const tableTile = tiles?.find(tile => tile.content.type === "CodapCaseTable")
      if (tableTile) { // a case table for the data set already exists in document
        uiState.setFocusedTile(tableTile.id)
        return
      }
    }
    // open a table and add it to the shared model manager
    openTableForDataset(model, caseMetadata)
```
It's essentially a two-by-two table where `openTableForDataSet` is being called in three of the four paths, so the early return in the fourth path allows us to eliminate the redundant calls.

Reply:
Was able to make the whole logic simpler by removing the check for existing tables.